ci-operator/step-registry/baremetalds/devscripts/gather/baremetalds-devscripts-gather-commands.sh

```diff
@@ -55,20 +31,65 @@ echo "Get install-gather, if there is one..."
 cp /root/dev-scripts/ocp/*/log-bundle*.tar.gz /tmp/artifacts/log-bundle-\$HOSTNAME.tar.gz || true
 
 echo "Get sosreport including sar data..."
-sos report --case-id "\$HOSTNAME" --batch -o container_log,filesys,kvm,libvirt,logs,networkmanager,podman,processor,rpm,sar,virsh,yum --tmp-dir /tmp/artifacts
+sos report --case-id "\$HOSTNAME" --batch \
+  -o container_log,filesys,kvm,libvirt,logs,networkmanager,podman,processor,rpm,sar,virsh \
+  -k podman.all -k podman.logs \
+  --tmp-dir /tmp/artifacts
 
 echo "Get libvirt logs..."
 tar -czC "/var/log/libvirt/qemu" -f "/tmp/artifacts/libvirt-logs-\$HOSTNAME.tar.gz" --transform "s?^\.?libvirt-logs-\$HOSTNAME?" .
 
+. common.sh
+. network.sh
+. utils.sh
+
+# Pass master IPs to installer-gather script to collect info from nodes which didn't join the cluster
+NODE_IPS=()
+for (( n=0; n<\$NUM_MASTERS; n++ ))
+do
+    node_name=\$(printf \$MASTER_HOSTNAME_FORMAT \$n)
+    node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//ip/dhcp[*]/host[@name='\${node_name}']/@ip)" -)
+    NODE_IPS+=("\$node_ip")
+done
+for (( n=0; n<\$NUM_WORKERS; n++ ))
+do
+    node_name=\$(printf \$WORKER_HOSTNAME_FORMAT \$n)
+    node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//dns[*]/host/hostname[. = '\$node_name']/../@ip)" -)
+    NODE_IPS+=("\$node_ip")
+done
+for (( n=0; n<\$NUM_EXTRA_WORKERS; n++ ))
+do
+    node_name="extraworker-%d"
+    node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//dns[*]/host/hostname[. = '\$node_name']/../@ip)" -)
+    NODE_IPS+=("\$(wrap_if_ipv6 \$node_ip)")
+done
+
+INTERNAL_SSH_OPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+)
+
+# Collect sos report from each known node
+for NODE_IP in \${NODE_IPS[@]}; do
+  echo "Fetching SOS report from \${NODE_IP}"
+  ssh "\${INTERNAL_SSH_OPTS[@]}" core@\${NODE_IP} sudo mkdir /run/artifacts &&
+  ssh "\${INTERNAL_SSH_OPTS[@]}" core@\${NODE_IP} \
+    sudo podman run -it --name toolbox --authfile /var/lib/kubelet/config.json --privileged --ipc=host --net=host --pid=host -e HOST=/host -e NAME=toolbox- -e IMAGE=registry.redhat.io/rhel8/support-tools:latest -v /run:/run -v /var/log:/var/log -v /etc/machine-id:/etc/machine-id -v /etc/localtime:/etc/localtime -v /:/host registry.redhat.io/rhel8/support-tools:latest \
```

Comment:
worth testing a pure-ipv6 case, as for test-infra we had to do this:
https://github.com/openshift/assisted-test-infra/pull/2067

or we can check it outside of this PR scope, I think it's only an improvement (as long as it's not causing other artifacts from being gathered)

Reply:
Yea this won't work on IPv6 which doesn't have outbound internet.  But, I wouldn't add internet outbound via proxy to IPv6 cluster since I think it's our only fully disconnected install in CI. I wouldn't want to lose that coverage.

```
Trying to pull registry.redhat.io/rhel8/support-tools:latest...
time="2023-03-22T11:00:19Z" level=warning msg="Failed, retrying in 1s ... (1/3). Error: initializing source docker://registry.redhat.io/rhel8/support-tools:latest: pinging container registry registry.redhat.io: Get \"https://registry.redhat.io/v2/\": dial tcp 23.11.208.241:443: connect: network is unreachable"
```